Configurations.md

```diff
@@ -2353,7 +2353,11 @@ Require a specific version of rustfmt. If you want to make sure that the
 specific version of rustfmt is used in your CI, use this option.
 
 - **Default value**: `CARGO_PKG_VERSION`
-- **Possible values**: any published version (e.g. `"0.3.8"`)
+- **Possible values**: `semver` compliant values, such as defined on [semver.org](https://semver.org/) and following the following [`semver::Version`](https://docs.rs/semver/latest/semver/struct.Version.html#total-ordering) specification.
+  - Multiple values can be used, they should be separated by a comma, e.g.: `1.0.0, 1.1.0`. The comparison is done using the `&&` operator, so all versions from a given range must be satisfied.
+  - By default, values without comparison operators are treated as `^` (caret) version requirements.
+  - If you want to specify an exact version, use the operator `=`, e.g.: `=1.0.0`.
+  - If you want to specify a range, use the operator `, `, e.g.: `1.0.0, <2.0.0`. This is the same as `^1.0.0, <2.0.0` or `^1.0.0`.
```

Comment:
These explanations are good, but I think it would be more valuable for users to see exactly how to specify these in their rustfmt.toml files. For consistency with the other options let's move this entire bulleted list below the `**Stable**` bullet. Having put a little more thought into this I think it might even be more useful to define subsections for the given semver targeting instead of bullets. Something like:

````markdown
### Specify a version range
```toml
required_version="1.0.0, <2.0.0"
```
````

>  By default, values without comparison operators are treated as `^` (caret) version requirements.

Two notes:
1. Users who aren't that familiar with semver might be confused by what the `^` means. Can we change this to something like: `Match on new minor or patch versions`.
2. I want to think a little more on the actual semantics of this. Right now I'm leaning towards keeping the existing behaviour where `required_version=1.0.0` means an exact match for 1.0.0. I know this means we'll need to workaround how the `semver` crate wants to parse the `VersionReq`. We can brainstorm different ideas for how to do that. let me know what your thoughts are on this.

Reply:
> I want to think a little more on the actual semantics of this. Right now I'm leaning towards keeping the existing behaviour where required_version=1.0.0 means an exact match for 1.0.0. I know this means we'll need to workaround how the semver crate wants to parse the VersionReq. We can brainstorm different ideas for how to do that. let me know what your thoughts are on this.

Agree with you! While I did use the standard `semver` behavior to avoid complexity, I wrote a draft matching the exact version: https://github.com/rust-lang/rustfmt/pull/6066/commits/73e75adc63e337c048cfdac380710b7dd1939ba7
Now that we split `check_semver_version`, it'll be easier to define this behavior.

Reply:
Changed the docs + the behavior. Let me know if it requires additional tweaks